src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -33,8 +33,10 @@ static Value builtinRegExpConstructor(ExecutionState& state, Value thisValue, si
     if (patternIsRegExp) {
```

Comment:
It seems that `builtinRegExpConstructor` is somewhat different from the [ES6 spec](https://www.ecma-international.org/ecma-262/6.0/#sec-regexp-constructor). Would you check it once more?

Reply:
Yes, it is different, but it's not the goal of this patch to fully upgrade the constructor to ES6 spec. I added the `#ifndef` because this PR would fail otherwise. Is it okay if I update the constructor and maybe other `RegExp` builtins in a separate patch?

Reply:
Yes, sure. Please update the constructor in other patch. Thanks.